lib/bap_llvm/config/llvm_configurator.ml

```diff
@@ -30,9 +30,13 @@ let args = [
 
 
 let () = C.main ~args ~name:"bap-llvm" @@ fun self ->
+  let linkmode =
+    "--link-" ^
+    (String.strip @@
+     C.Process.run_capture_exn self llvm_config ["--shared-mode"]) in
```

Comment:
I wonder if the `--shared-mode` command-line option is present on all llvm versions that we support? Maybe, it is better to be more defensive here (both against previous and future versions) and anticipate possible failures by sticking with the static mode by default.

Reply:
It exists at least as far back as llvm 4.0. I don't know if there's a great solution here in general, since the installed version of llvm could change since bap was built, and then we might have a mismatch of the linking mode anyway.